src/redux/data.ts

```diff
@@ -1399,61 +1422,58 @@ export const dataAddNewTransaction = (
   dispatch: ThunkDispatch<
     AppState,
     unknown,
-    DataAddNewTransactionSuccessAction
+    DataUpdatePendingTransactionSuccessAction
   >,
   getState: AppGetState
-) =>
-  withRunExclusive(
-    async (): Promise<RainbowTransaction | undefined> => {
-      const { transactions } = getState().data;
-      const { accountAddress, nativeCurrency, network } = getState().settings;
-      if (
-        accountAddressToUpdate &&
-        toLower(accountAddressToUpdate) !== toLower(accountAddress)
-      )
-        return;
-      try {
-        const parsedTransaction = await parseNewTransaction(
-          txDetails,
-          nativeCurrency
-        );
-        const _transactions = [parsedTransaction, ...transactions];
-        dispatch({
-          payload: _transactions,
-          type: DATA_ADD_NEW_TRANSACTION_SUCCESS,
-        });
-        saveLocalTransactions(_transactions, accountAddress, network);
-        if (parsedTransaction.from && parsedTransaction.nonce) {
-          await dispatch(
-            incrementNonce(
-              parsedTransaction.from,
-              parsedTransaction.nonce,
-              parsedTransaction.network
-            )
-          );
-        }
-        if (
-          !disableTxnWatcher ||
-          network !== Network.mainnet ||
-          parsedTransaction?.network
-        ) {
-          dispatch(
-            watchPendingTransactions(
-              accountAddress,
-              parsedTransaction.network
-                ? TXN_WATCHER_MAX_TRIES_LAYER_2
-                : TXN_WATCHER_MAX_TRIES,
-              null,
-              // @ts-expect-error `watchPendingTransactions` only takes 3 arguments.
-              provider
-            )
-          );
-        }
-        return parsedTransaction;
-        // eslint-disable-next-line no-empty
-      } catch (error) {}
+) => {
+  const { pendingTransactions } = getState().data;
+  const { accountAddress, nativeCurrency, network } = getState().settings;
+  if (
+    accountAddressToUpdate &&
+    toLower(accountAddressToUpdate) !== toLower(accountAddress)
+  )
+    return;
+  try {
+    const parsedTransaction = await parseNewTransaction(
+      txDetails,
+      nativeCurrency
+    );
+    const _pendingTransactions = [...pendingTransactions, parsedTransaction];
```

Comment:
the ordering of this has switched from "existing pending txns, then new txn" - presume we want `[parsedTransaction, ...pendingTransactions]`